src/Docfx.Common/ConvertToObjectHelper.cs

```diff
@@ -1,36 +1,65 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
 using System.Dynamic;
 
 using Newtonsoft.Json.Linq;
+using System.Runtime.CompilerServices;
 
 namespace Docfx.Common;
 
 public static class ConvertToObjectHelper
 {
-    public static object ConvertExpandoObjectToObject(object raw)
+#nullable enable
+    [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+    public static object? ConvertJObjectToObject(object raw)
     {
-        return ConvertExpandoObjectToObjectCore(raw, new Dictionary<object, object>());
+        switch (raw)
+        {
+            case null:
+                return null;
+            case JToken jToken:
+                return ConvertJTokenToObject(jToken);
+            default:
+                return raw; //  if other type object passed. Return object itself.
+        }
     }
 
-    public static object ConvertJObjectToObject(object raw)
+    [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+    private static object? ConvertJTokenToObject([NotNull] JToken jToken)
     {
-        if (raw is JValue jValue)
-        {
-            return jValue.Value;
-        }
-        if (raw is JArray jArray)
-        {
-            return jArray.Select(ConvertJObjectToObject).ToArray();
-        }
-        if (raw is JObject jObject)
+        switch (jToken.Type)
         {
-            return jObject.ToObject<Dictionary<string, object>>().ToDictionary(p => p.Key, p => ConvertJObjectToObject(p.Value));
+            case JTokenType.Array:
+                return ConvertJArrayToObjectArray((JArray)jToken);
+            case JTokenType.Object:
+                return ConvertJObjectToDictionary((JObject)jToken);
+            default:
+                if (jToken is JValue jValue)
```

Comment:
nit: why not also cast here?
```suggestion
                return ((JValue)jToken).Value;
```